samples/tiled-convolution/tiled-convolution.cpp

```diff
@@ -123,23 +122,72 @@ void inline compute_index(const index_t total_size_dim,
   }
 }
 
+template <typename event_t, typename start_t>
+void inline profiler(event_t& events, const start_t& starts) {
```

Comment:
Isn't this just std::vector<event_t>? Why is it a template when we know what it is?